### PR TITLE
Vet clinic database: query and update animals table

### PR DESCRIPTION
**Pull request by Mikey1-nuel**
## Vet clinic database: query and update animals table

**In this project I:**

- Added a column `species` of type `string` to my `animals` table and modified my `schema.sql` file.
- Inserted the following data:
  - Animal: His name is Charmander. He was born on Feb 8th, 2020, and currently weighs -11kg. He is not neutered and he has never tried to escape.
  - Animal: Her name is Plantmon. She was born on Nov 15th, 2021, and currently weighs -5.7kg. She is neutered and she has tried to escape 2 times.
  - Animal: His name is Squirtle. He was born on Apr 2nd, 1993, and currently weighs -12.13kg. He was not neutered and he has tried to escape 3 times.
  - Animal: His name is Angemon. He was born on Jun 12th, 2005, and currently weighs -45kg. He is neutered and he has tried to escape once.
  - Animal: His name is Boarmon. He was born on Jun 7th, 2005, and currently weighs 20.4kg. He is neutered and he has tried to escape 7 times.
  - Animal: Her name is Blossom. She was born on Oct 13th, 1998, and currently weighs 17kg. She is neutered and she has tried to escape 3 times.
  - Animal: His name is Ditto. He was born on May 14th, 2022, and currently weighs 22kg. He is neutered and he has tried to escape 4 times.
- Added these insertions to my `data.sql` file.

- Inside a transaction, I updated the `animals` table by setting the `species` column to `unspecified`. Verified that change was made. Then rolled back the change and verified that the `species` columns went back to the state before the transaction.
- Inside a transaction I:
  - Updated the `animals` table by setting the `species` column to `digimon` for all animals that have a name ending in `mon`.
  - Updated the `animals` table by setting the `species` column to `pokemon` for all animals that don't have `species` already set.
  - Verified that changes were made.
  - Committed the transaction.
  - Verified that changes persist after commit.
- I took a screenshot of the results of my actions.
- Inside a transaction, I deleted all records in the `animals` table, then rolled back the transaction.
- After the rollback, I verified if all records in the `animals` table still exist.
- I took a screenshot of the results of my actions.
- Inside a transaction I:
  - Deleted all animals born after Jan 1st, 2022.
  - Created a savepoint for the transaction.
  - Updated all animals' weight to be their weight multiplied by -1.
  - Rollbacked to the savepoint.
  - Updated all animals' weights that were negative to be their weight multiplied by -1.
  - Committed the transaction.
- I took a screenshot of the results of my actions.**
- Added all these transactions to my `queries.sql` file.

- I wrote queries to answer the following questions:
  - How many animals were there?
  - How many animals had never tried to escape?
  - What was the average weight of the animals?
  - Who escaped the most, neutered or not neutered animals?
  - What was the minimum and maximum weight of each type of animal?
  - What was the average number of escape attempts per animal type of those born between 1990 and 2000?
- I took a screenshot of the results of your queries.
- I added these queries to my `queries.sql` file.

Screenshot (394)
Screenshot (387)
Screenshot (388)
Screenshot (390)
Screenshot (392)
Screenshot (393)

### CODE REVIEW

File: data.sql [hunk 1]
```diff
@@ -11,3 +11,24 @@ INSERT INTO animals (name, date_of_birth, escape_attempts, neutered, weight_kg)
 
 INSERT INTO animals (name, date_of_birth, escape_attempts, neutered, weight_kg)
     VALUES ('Devimon', '2017-05-12', 5, true, 11);
+
+INSERT INTO animals (name, date_of_birth, escape_attempts, neutered, weight_kg)
+    VALUES ('Charmander', '2020-02-08', 0, false, 11);
```
Comment (bobb-Rob) [inline data.sql]: - [ ] From the data provided in the project requirement document, the weight of `Charmander` is a negative decimal `-11.0` value. Kindly update your code.

File: data.sql [hunk 2]
```diff
@@ -11,3 +11,24 @@ INSERT INTO animals (name, date_of_birth, escape_attempts, neutered, weight_kg)
...
+
+INSERT INTO animals (name, date_of_birth, escape_attempts, neutered, weight_kg)
+    VALUES ('Plantmon', '2021-11-15', 2, true, 5.7);
```
Comment (bobb-Rob) [inline data.sql]: - [ ] From the data provided in the project requirement document, the weight of `Plantmon` is a negativevalue with decimal `-5.7` . Kindly update your code.

File: data.sql [hunk 3]
```diff
@@ -11,3 +11,24 @@ INSERT INTO animals (name, date_of_birth, escape_attempts, neutered, weight_kg)
...
+
+INSERT INTO animals (name, date_of_birth, escape_attempts, neutered, weight_kg)
+    VALUES ('Squirtle', '1993-04-02', 3, false, 12.13);
```
Comment (bobb-Rob) [inline data.sql]: - [ ] From the data provided in the project requirement document, the weight of `Squirtle` is a negative value with decimal `-12.13` . Kindly update your code.

File: data.sql [hunk 4]
```diff
@@ -11,3 +11,24 @@ INSERT INTO animals (name, date_of_birth, escape_attempts, neutered, weight_kg)
...
+
+INSERT INTO animals (name, date_of_birth, escape_attempts, neutered, weight_kg)
+    VALUES ('Angemon', '2005-06-12', 1, true, -45);
```
Comment (bobb-Rob) [inline data.sql]: Angemon has a decimal value of `45.0`. Please update your code.

**Review by bobb-Rob — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️
Hi @Mikey1-nuel, 

Good job so far! 👏👏👏 
There are some issues that you still need to work on to go to the next project but you are almost there!  😃 
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>


## Highlights
✅ No Linter error
✅ Correct use of Gitflow
✅ Very descriptive Pull Request(PR)
✅ Neat Commit messages

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Qoosim — APPROVED** [review]
#  *Status: Approved* :tada: :heavy_check_mark: 

Hi @Mikey1-nuel, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
-  Good README file. :heavy_check_mark: 
-  Correct Gitflow. :heavy_check_mark: 



## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**